runtime-modules/content/src/nft/mod.rs

```diff
@@ -283,51 +283,54 @@ impl<T: Trait> Module<T> {
     /// Buy nft
     pub(crate) fn buy_now(
         in_channel: T::ChannelId,
-        mut nft: Nft<T>,
-        owner_account_id: T::AccountId,
+        nft: Nft<T>,
+        old_owner_account_id: Option<T::AccountId>,
         new_owner_account_id: T::AccountId,
         new_owner: T::MemberId,
     ) -> Nft<T> {
         if let TransactionalStatus::<T>::BuyNow(price) = &nft.transactional_status {
-            Self::complete_payment(
-                in_channel,
-                nft.creator_royalty,
-                *price,
-                new_owner_account_id,
-                Some(owner_account_id),
-                false,
-            );
-
-            nft.owner = NftOwner::Member(new_owner);
+            if old_owner_account_id.is_some() {
+                Self::complete_payment(
+                    in_channel,
+                    nft.creator_royalty,
+                    price.to_owned(),
+                    new_owner_account_id,
+                    old_owner_account_id,
+                    false,
+                );
+            } else {
+                let _ = Balances::<T>::slash(&new_owner_account_id, price.to_owned());
+            }
         }
 
-        Nft::<T> {
-            transactional_status: TransactionalStatus::<T>::Idle,
-            ..nft
-        }
+        nft.with_transactional_status(TransactionalStatus::<T>::Idle)
+            .with_member_owner(new_owner)
     }
 
     /// Completes nft offer
     pub(crate) fn complete_nft_offer(
         in_channel: T::ChannelId,
         mut nft: Nft<T>,
-        owner_account_id: T::AccountId,
+        owner_account_id: Option<T::AccountId>,
         new_owner_account_id: T::AccountId,
     ) -> Nft<T> {
         if let TransactionalStatus::<T>::InitiatedOfferToMember(to, price) =
             &nft.transactional_status
         {
             if let Some(price) = price {
-                Self::complete_payment(
-                    in_channel,
-                    nft.creator_royalty,
-                    *price,
-                    new_owner_account_id,
-                    Some(owner_account_id),
-                    false,
-                );
+                if owner_account_id.is_some() {
+                    Self::complete_payment(
```

Comment:
Can you add a comment here about what the scenario is, why there is this flow of royalites, proceeds and fee.